flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/DeleteFromFilterOperation.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.operations;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.connector.sink.abilities.SupportsDeletePushDown;
+import org.apache.flink.table.expressions.ResolvedExpression;
+
+import java.util.Collections;
+import java.util.List;
+
+/** The operation for deleting data in a table according to filters directly. */
+@Internal
+public class DeleteFromFilterOperation implements Operation {
+    private final SupportsDeletePushDown supportsDeletePushDown;
+    private final List<ResolvedExpression> filters;
+
+    public DeleteFromFilterOperation(
+            SupportsDeletePushDown supportsDeletePushDown, List<ResolvedExpression> filters) {
```

Comment:
add `Nonnull` annotation and null check since only accept non-empty objects when do conversion